### PR TITLE
Header and footer design updates

### PR DESCRIPTION
**Pull request by davidrapson**
**Header** (NP-1024)

- Fixes spacing between header and navigation
- Search input now takes up 6 columns

| Before | After | 
|---|---|
| image | image |

**Footer** (NP-1033)

Small spacing fixes, but notably fixes wrapping on small screens

| Before | After | 
|---|---|
| image | image |

**Removes logo and website_feedback as standalone components**

The are never used outside of the header and footer so shouldn't really be included as standalone components. By inlining them is also makes some of the `locals` passing simpler as there's fewer layers to pass through.

**Allows more granular prepend_view_path**

The above change also supports a more granular `prepend_view_path` such as `prepend_view_path(Rails.root.join("node_modules/@citizensadvice"))`. @pedantic-git pointed out that setting the view path too wide means that Rails has to look through the _whole_ of `node_modules` when rendering a page. Being able to narrow the scope to our namespace avoids this and means that you can write the shorter `= render partial: "design-system/haml/callout"` in templates.

Previously our haml components assumed a certain view path and that `render` was available inside our published partials. This makes changing the view path difficult. As it happens restricting our `haml/` files to being standalone is actually pretty reasonable and removes this constraint.

**Note**: This problem might go away in future if we choose to publish as a gem or rails engine

### CODE REVIEW

**Review by pedantic-git — COMMENTED** [review]
I can't really hit the approve button because I don't know the code well enough but this looks great!

Where is the fix that allows the tighter prefix for `render partial`? I couldn't immediately see it in the diff.

File: haml/_footer.html.haml [hunk 1]
```diff
@@ -1,80 +1,83 @@
...
-  = render "@citizensadvice/design-system/haml/website_feedback"
+  %p.cads-footer__feedback
+    %a{ href: "https://www.research.net/r/J8PLH2H", target: "_blank", rel: "noopener" }= t("cads.footer.website_feedback")
+    %span.cads-footer__feedback-icon.cads-icon_external-link
   .cads-grid-container
     .cads-grid-row
       .cads-grid-col-md-3
-        %h2 Advice
-        %ul
+        %h2.cads-footer__section-title Advice
```
[the remark this answers is not in the document]
Comment (davidrapson) [reply haml/_footer.html.haml:8]: Nope, they are the last thing we need to sort. @baseonmars is looking at it as part of updating the header and footer for advisernet as far as I know so haven't tackled that in this PR.

**Comment by davidrapson** [comment]
> Where is the fix that allows the tighter prefix for render partial? I couldn't immediately see it in the diff.

The fix is removing any use of `render` from _within_ components in `haml/` so any component we distribute are standalone. So anywhere where I am removing `= render` in the diff.

File: haml/_footer.html.haml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pedantic-git) [reply haml/_footer.html.haml:8]: Cool!

[the remark this answers is not in the document]
Comment (baseonmars) [reply haml/_footer.html.haml:8]: Tickets NP-662/NP-761 - both will be completed this sprint.

File: scss/6-components/_header.scss [hunk 2]
```diff
@@ -22,52 +23,57 @@
...
       display: none;
     }
 
-    .cads-search {
-      margin-top: 0;
-    }
-
-    .cads-header-search-row {
+    .cads-header__search-row {
       margin-top: 1rem;
```
Comment (baseonmars) [inline scss/6-components/_header.scss:66]: may as well change this to a variable whilst you're in here.

**Review by baseonmars — APPROVED** [review]
Looks great - one small thing I noticed but no need to hold this back.

**Pull request merged**